homeassistant/components/edl21/sensor.py

```diff
@@ -292,8 +292,8 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the EDL21 sensor."""
-    hass.data[DOMAIN] = EDL21(hass, config_entry.data, async_add_entities)
-    await hass.data[DOMAIN].connect()
+    api = EDL21(hass, config_entry.data, async_add_entities)
+    await api.connect()
 
 
 class EDL21:
```

Comment:
Note: I feel this EDL21 class should be moved outside of `sensor.py`, possibly into `__init__.py`, and use a separate dispatcher for new entities.

That is out of scope for this PR, and should involve the code owner.

Reply:
cc @StephanU who implemented the config-flow